rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -98,6 +98,27 @@ module Impl {
 
     /** Gets an access to this variable. */
     VariableAccess getAnAccess() { result.getVariable() = this }
+
+    /**
+     * Gets the pattern that declares this variable.
+     *
+     * Normally, the pattern is unique, except when introduced in an or pattern:
+     *
+     * ```rust
+     * match either {
+     *    Either::Left(x) | Either::Right(x) => println!(x),
+     * }
+     * ```
```

Comment:
So in that case we would get two results for `getPat`, namely the two `x`s on the left-hand side of the `=>`?

Reply:
Yes.